sos/report/plugins/microovn.py

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2024 Alan Baghumian <alan.baghumian@canonical.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, UbuntuPlugin
+
+
+class MicroOVN(Plugin, UbuntuPlugin):
+    """The MicroOVN plugin collects the current status of the microovn
+    snap.
+
+    It will collect journald logs as well as output from various microovn
+    commands.
+    """
+
+    short_desc = 'MicroOVN Snap'
+    plugin_name = "microovn"
+    profiles = ('network', 'virt')
+
+    packages = ('microovn', )
+    commands = ('microovn', )
+
+    def setup(self):
+        self.add_journal(units="snap.microovn.*")
+
+        microovn_subcmds = [
+            'cluster list',
+            'status',
+            'certificates list',
```

Comment:
Does this show the certificates in raw format at all, and therefore need to obfuscate?

Reply:
No obfuscation needed. This is the output:

```
$ sudo microovn certificates list
[OVN CA]
/var/snap/microovn/common/data/pki/cacert.pem (OK: Present)

[OVN Northbound Service]
/var/snap/microovn/common/data/pki/ovnnb-cert.pem (OK: Present)
/var/snap/microovn/common/data/pki/ovnnb-privkey.pem (OK: Present)

[OVN Southbound Service]
/var/snap/microovn/common/data/pki/ovnsb-cert.pem (OK: Present)
/var/snap/microovn/common/data/pki/ovnsb-privkey.pem (OK: Present)

[OVN Northd Service]
/var/snap/microovn/common/data/pki/ovn-northd-cert.pem (OK: Present)
/var/snap/microovn/common/data/pki/ovn-northd-privkey.pem (OK: Present)

[OVN Chassis Service]
/var/snap/microovn/common/data/pki/ovn-controller-cert.pem (OK: Present)
/var/snap/microovn/common/data/pki/ovn-controller-privkey.pem (OK: Present)

[Client]
/var/snap/microovn/common/data/pki/client-cert.pem (OK: Present)
/var/snap/microovn/common/data/pki/client-privkey.pem (OK: Present)
```